src/controllers/update/company.profile.controller.ts

```diff
@@ -0,0 +1,53 @@
+import { Session } from "@companieshouse/node-session-handler";
+import { NextFunction, Request, Response } from "express";
+import { ApplicationData } from "model";
+import { logger } from "../../utils/logger";
+import * as config from "../../config";
+import { getApplicationData, setExtraData } from "../../utils/application.data";
+import { getCompanyRequest } from "../../service/overseas.entities.service";
+import { OeErrorKey } from "../../model/data.types.model";
+import { mapCompanyProfileToOverseasEntityToDTOUpdate } from "../../utils/update/company.profile.mapper.to.oversea.entity";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `GET ${config.CONFIRM_OVERSEAS_ENTITY_DETAILS_PAGE}`);
+    const session = req.session as Session;
+    const appData: ApplicationData = getApplicationData(session);
+    const id: string = appData?.oe_number || "";
+    const companyDataResponse = await getCompanyRequest(req, id);
```

Comment:
cast to CompanyProfile from node api sdk
await getCompanyRequest(req, id) as CompanyProfile